### PR TITLE
Add admin & dev route guards

### PR DESCRIPTION
**Pull request by ibolton336**
- Adds roles files for defining available permissions for routes & components
- Implements reusable logic for pulling user roles and scopes for reuse across app 
- Implements route guarding for dev/admin routes
- Add RBAC implementation for read/write permissions
Blocked:
- Waiting on confirmation of wether or not we will need assessment permissions for migrator to read but not execute assessments against pathfinder api.

### CODE REVIEW

**Review by gildub — COMMENTED** [review]
Still running tests, meanwhile here's my initial review, more to follow

File: pkg/client/src/app/common/RouteWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+export const RouteWrapper = ({
+  Component,
+  roles,
+  ...props
+}: IRouteWrapperProps) => {
+  const [parsedToken, setParsedToken] = useLocalStorageContext(
+    LocalStorageKey.parsedToken
+  );
+  const token = (parsedToken && JSON.parse(parsedToken)) || undefined;
+  let userRoles: Array<any> = token?.realm_access?.roles,
```
Comment (gildub) [inline pkg/client/src/app/common/RouteWrapper.tsx:23]: Why use `let` here ?
Also the reaml_access isn't an array of string ?

File: pkg/client/src/app/common/RouteWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    //TODO: Implement granular user permissions
+    scope: string = token?.scope,
```
Comment (gildub) [inline pkg/client/src/app/common/RouteWrapper.tsx:25]: scope doesn't seem used.

**Review by gildub — COMMENTED** [review]
Just re-reviewed some of the types, please see inline.

File: pkg/client/src/app/common/RouteWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  LocalStorageKey,
+  useLocalStorageContext,
+} from "@app/context/LocalStorageContext";
+import React from "react";
+import { Redirect, Route } from "react-router-dom";
+import { checkAccess } from "./rbac-utils";
+
+interface IRouteWrapperProps {
+  Component: React.ComponentType<any>;
+  roles: Array<string>;
```
[the remark this answers is not in the document]
Comment (ibolton336) [reply pkg/client/src/app/common/RouteWrapper.tsx:11]: +1

**Review by gildub — APPROVED** [review]
Good enough to be merge and keep going forward.

**Review by gildub — APPROVED** [review]
LGTM

**Pull request merged**